c10/cuda/CMakeLists.txt

```diff
@@ -0,0 +1,58 @@
+# Build file for the C10 CUDA.
+#
+# C10 CUDA is a minimal library, but it does depend on CUDA.
+
+include(../../cmake/public/utils.cmake)
+include(../../cmake/public/cuda.cmake)
+
+# ---[ Configure macro file.
+set(C10_CUDA_BUILD_SHARED_LIBS ${BUILD_SHARED_LIBS}) # used in cmake_macros.h.in
+# Probably have to do this :(
+configure_file(
+    ${CMAKE_CURRENT_LIST_DIR}/impl/cuda_cmake_macros.h.in
+    ${CMAKE_BINARY_DIR}/c10/cuda/impl/cuda_cmake_macros.h)
+
+# Note: if you want to add ANY dependency to the c10 library, make sure you
+# check with the core PyTorch developers as the dependendency will be
+# transitively passed on to all libraries dependent on PyTorch.
+file(GLOB C10_CUDA_SRCS
+        *.cpp
+        *.cu
+        impl/*.cpp
+        impl/*.cu
+        )
+file(GLOB_RECURSE C10_CUDA_ALL_TEST_FILES test/*.cpp)
```

Comment:
Let's not introduce a new glob, but immediately list the source files. We want to move other targets there as well.

Reply:
Oh right. I had just copied this from c10/CMakeLists.txt and forgot that you're signed up to de-glob that cmake.